### ISSUE TITLE
First sentence after defgroup no longer used as brief description for sub-groups

### ISSUE THREAD

**Issue by omarcou** [doxygen/doxygen#10569]
**Describe the bug**
In Doxygen 1.10.0, the option JAVADOC_AUTOBRIEF does not seem to work any more.
This is a problem because it forces us to modify a large number of files by explicitly set their brief description in order to have the same result as before.

**Expected behavior**
When using the following code:
```
/**
* @defgroup MyGroupName My group name
* @ingroup MyParentGroupName
*
* Text for brief description. Text for long description.
*/
```
The result with JAVADOC_AUTOBRIEF activated should be (was for Doxygen 1.9) that in the page of MyParentGroupName, the subgroup MyGroupName would appear with 'Text for brief description' as brief description.

**Actual behavior**
Only the name of subgroup MyGroupName appears with the code above, there is no brief description.

**To Reproduce**
Run the bat script (after editing it to set the location of doxygen on your machine) inside the following reproducer and check the result in the appropriate docgenXX subfolder. The result for page Modules/topics.html will not be the same depending on which version of Doxygen was selected in the bat script. 
[reproBugGroupBrief.zip](https://github.com/doxygen/doxygen/files/13987295/reproBugGroupBrief.zip)

**Version**
Doxygen 1.10.0, Windows 10.

**Comment by albert-github** [comment]
I think also the content of the settings file is of importance her and this is not provided at all, so in total:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by omarcou** [comment]
I joined a reproducer and also corrected the first message, because making the reproducer made realize what the bug actually was: it is just the JAVADOC_AUTOBRIEF option that seems to be ignored now.
I checked the changelog and I noticed that there was a fix regarding this option. It may have broken something.

**Comment by albert-github** [comment]
Regarding the `JAVADOC_AUTOBRIEF` you mention:
> I checked the changelog and I noticed that there was a fix regarding this option. It may have broken something.

which fix did you mean here (probably one regarding `\ifile` and `\iline` presence in the brief description so it is not empty anymore, which now shows up with the grouping commands as well).

**Comment by omarcou** [comment]
That was actually a mistake, I did not see that the first occurrence of 'JAVADOC_AUTOBRIEF' in the changelog page was not in the section dedicated to the version 1.10.0. I think the bugfix I saw was in some 1.9.x version. Which makes it quite weird for JAVADOC_AUTOBRIEF to be broken in 1.10.0...

**Comment by olivier-marcou** [comment]
I do not see any recent update on this issue, but regardless I tested the latest 1.12.0 release of Docygen, and I found out that it is partially solved.

One may even consider it is totally solved strictly speaking, since for the specific example I gave in the issue description and the reproducer, the bug does not happen any more: if a @defgroup is followed by an @ingroup which is followed by a description, the first sentence of said description will effectively now be taken as brief description.

However, when the @defgroup is not followed by @ingroup (so, in the case of a top-level group), the bug still happens and the first sentence of description will not be used as brief description.

This case was not explicitly covered by the original reproducer, as we had not included a description for the parent group in file includes/groups.txt, however it is quite easy to modify the said reproducer to include that case.

Should I open a new issue and/or post a new reproducer for this specific unfixed case?

**Comment by albert-github** [comment]
Thanks for the feedback.

> Should I open a new issue and/or post a new reproducer for this specific unfixed case?

I think best is to add the reproducer here.

**Comment by albert-github** [comment]
Regarding the comment I don't see the reproducer yet...

I tried the following scenarios (with doxygen 1.12.0):
```
/**
* @defgroup MyParentGroupName_1 Parent group name brief
* Text for brief description.
*/

/**
* @defgroup MyParentGroupName_2 Parent group name empty brief
*
* Text for brief description.
*/

/**
* Text for brief description.
* @defgroup MyParentGroupName_3 Brief Parent group name
*/

/**
* Text for brief description.
*
* @defgroup MyParentGroupName_4 Brief Empty Parent group name
*/
```
![image](https://github.com/user-attachments/assets/4a5bae55-1c64-49ff-8402-1a83cc9645d7)

So it looks to me that all but `MyParentGroupName_2` result in a brief description 

Example: [example.tar.gz](https://github.com/user-attachments/files/17230128/example.tar.gz)

**Comment by olivier-marcou** [comment]
Thanks for looking into the matter.
I had not looked into all the possibilities that you tested there.
Also, I just tried getting a second reproducer ready, but then I realized that the first one already does the job: the only things that changed between both reproducers is how I name some variables/files/folders (because of switching from 1.10 to 1.12), the path to the 1.12 doxygen and obvisously the results, but the reproducer stays functionally the same.

**Comment by albert-github** [comment]
Looks like the fact that the fact that `MyParentGroupName_2` doesn't show a brief  text is a regression and shows the first time in version 1.10.0.
More precise is a result from the fix (4729e92540dbf6eb971fbec73f8fc855bb84a0b7):
```
Commit: 4729e92540dbf6eb971fbec73f8fc855bb84a0b7 [4729e92]

Date: Wednesday, November 15, 2023 12:18:12 PM

Issue #10414 Report warning with wrong file and wrong line number

As the `addtogroup` parts are joined together in one block the second block didn't know anymore wheer it came from originally which resulted in the wrong file name and line number.
By adding some internal commands dureng provessing this has been corrected.
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11168

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
